irobotclient/configuration_handler.py

```diff
@@ -0,0 +1,159 @@
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import argparse
+import string
+import re
+import os
+import errno
+
+from irobotclient.custom_exceptions import IrobotClientException
+
+
+def _get_command_line_agrs():
+    """
+    Get program arguments, calls validation methods, and returns the values in the argparse object.
+
+    :return: args
+    """
+
+    parser = argparse.ArgumentParser(prog="irobot-client",
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     description="Command line interface for iRobot HTTP requests",
+                                     usage="irobot-client [options] INPUT_FILE OUTPUT_FILE")
+    parser.add_argument("input_file", help="path and name of input file")
+    parser.add_argument("output_dir", help="path of output directory")
+    parser.add_argument("-u", "--url", help="URL scheme, domain and port for irobot. EXAMPLE: http://irobot:5000/")
+    parser.add_argument("-t", "--token", help="Arvados authentication token")
+    parser.add_argument("-f", "--force", default=False, action="store_true", help="force overwrite output file if "
+                                                                                  "it already exists")
+    parser.add_argument("--no_index", default=False, action="store_true", help="Do not download index files for"
+                                                                               "CRAM/BAM files")
+    args = parser.parse_args()
+
+    print(f'_get_command_line_args() args: {args}')  # Beth - debug
+
+    return args
+
+
+def _validate_command_line_args(args):
+    """
+    Ensures all the necessary details are set to form the requests.
+
+    :param args:
+    :return:
+    """
+    _check_input_file_argument(args)
+    _check_output_directory_argument(args)
+    _check_url_argument(args)
+    _check_authorisation_token(args)
+
+
+def _check_input_file_argument(args):
+    """
+    Strip leading slash from input argument to prevent double slashes in API request.
+
+    :param args: the command line arguments
+    """
+    if args.input_file.endswith('/') or os.path.isdir(args.input_file):
+        raise IrobotClientException(errno=errno.ECONNABORTED,
+                                    message="Cannot download entire directories at present.")
+
+    if args.input_file.startswith('/'):
+        args.input_file = args.input_file.lstrip('/')
+
+
+def _check_output_directory_argument(args):
+    """
+    Check if the output directory already exists and if it already contains files of the same name as the input file.
+
+    :return: n/a
+
+    """
+
+    try:
```

Comment:
That's a good point.  I was doing a blanket try:catch but it's more deceptive than it is helpful really.